workshops/python_stock_viewer/README.md

```diff
@@ -0,0 +1,254 @@
+
+## Importing the Libraries
+
+For this program, we'll be using two libraries, the [Requests](https://requests.readthedocs.io/en/master/) library to handle our HTTP requests and the [JSON](https://docs.python.org/3/library/json.html) library to parse the values we get back from those requests into values that our code can understand.
+
+Both these libraries are set up for us out of the box, so we won't need to install them. So in your code, add these lines to the top of the files.
+
+```python
+import requests
+import json
+```
+These lines allow your code to access external libraries, in this case JSON and Requests, and work with them. They essentially extend Python's functionality and allow us to do more things with our code. We import these libraries because we want the functionality that these libraries offer.`import [LIBRARY-NAME]` is the syntax used to import the library `[LIBRARY-NAME]` in Python.
+
+
+## Setting up the API
+
+The first thing we need to do is to set up the API we'll be requesting from. API stands for "Application Programming Interface" and they basically allow you to access data. In order to access stock data, we'll need an API that provides this data. For this demonstration, we'll use the [Finnhub](https://finnhub.io/) stock API to get values.
+
+Of course, often times, APIs won't let you have the data for free. You typically need to have some method to prove your identity to the API to tell the API that you are allowed to make the request. This API requires an API key. This is a value that we pass to the API to prove that we are authenticated, or allowed to use the API. Luckily, Finnhub provides access to its API for free, so we'll need to go to their site and get one.
+
+Once you're on the site, you'll see this screen:
+
+![Screen](https://cloud-8ws90xn6j.vercel.app/image.png)
+
+Click on "Get free API key" and create an account. You should see this:
+
+![Key](https://cloud-ks0dqz7ba.vercel.app/image.png)
+
+There should be a key (I've hidden mine for security purposes - yours should be where the purple box is) but that's the key you'll need to use for authentication.
+
+Now let's set up the key in Python. The Finnhub API wants the key passed in through the headers, which are ways to send additional information. The Requests library allows you pass headers into the web request, which is how we'll authenticate. Let's define the headers.
+
+```python
+header = {'X-Finnhub-Token':'[TOKEN]'} 
+# Replace [TOKEN] with the token you got from Finnhub
+```
+How does this code work?  
+
+The line creates a Python dictionary with an attribute `'X-Finnhub-Token'` and a value of your token. This dictionary is saved as a variable, and later, we'll pass the variable into the request. When the server sees that your request has a token, it *authenticates* you, or lets you use the API.
+
+
+## Making our First Request
+
+So let's see if it works. Try running this code.
```

Comment:
Add something like “by clicking the green “Run” button at the top of the repl.it window”